### PR TITLE
Like&Dislike for Post&Comment

### PR DESCRIPTION
**Pull request by tayyabali12345**

### CODE REVIEW

**Comment by tiawana** [comment]
Use proper name for PR and don't use single commit for multiple file changes. and commit msg should be specific to the changes. 
@tayyabali12345

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -47,7 +47,29 @@ def destroy
...
+  end
+
+  def dislike
+    toogle_reaction(false)
+  end
+
   private
+
+    def toogle_reaction(reaction_type)
+      @comment = Comment.find(params[:id])
```
Comment (tiawana) [inline app/controllers/comments_controller.rb]: I think we already have a method for this

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -47,7 +47,29 @@ def destroy
...
+      existing_reaction = Reaction.find_by(user_id: current_user.id, likeable_id: @comment.id)
```
Comment (tiawana) [inline app/controllers/comments_controller.rb]: its gives you one record use where if needed more than one record

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -47,7 +47,29 @@ def destroy
...
+      if existing_reaction && existing_reaction.like == reaction_type
+        existing_reaction.destroy
+      elsif existing_reaction==nil
+        @comment.reactions.create(user_id: current_user.id, like: reaction_type)
+      else
+        @comment.reactions.update(like: reaction_type)
+      end
+      redirect_to post_path(@comment.post)
```
Comment (tiawana) [inline app/controllers/comments_controller.rb]: don't use redirect here.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: Other method is for Post reaction.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: I need only one.

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: Can you please elaborate?

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: Done.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: Done.

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (tayyabali12345) [reply app/controllers/comments_controller.rb]: Done.

**Pull request merged**